modules/dnn/src/layers/einsum_layer.cpp

```diff
@@ -0,0 +1,1178 @@
+
+    int64_t outputDimCounter = 0;
+    for (auto letter : rhs_eq)
+    {
+        if(letter == '.')
+        {
+            CV_Error(Error::StsNotImplemented, "Ellipsis are not supported yet");
+        } else {
+            if (middleOfEllipsis)
+            {
+                CV_Error(Error::StsError, "Encountered '.' character that is"
+                " not part of output subscript");
+            }
+
+            auto letterIndex = letterToIndex(letter);
+
+            if (letterIndex == -1)
+            {
+                 CV_Error(Error::StsError,
+                    "The only permissible subscript labels are"
+                    " lowercase letters (a-z) and uppercase letters (A-Z).");
+            }
+
+            if (outputLetterToCount[letterIndex] != 0)
+            {
+                CV_Error(Error::StsError,
+                 "Output subscript constains repeated letters");
+            }
+
+            ++outputLetterToCount[letterIndex];
+
+            auto mappedIndex = letter2index[letterIndex];
+
+            if(mappedIndex == -1)
+            {
+                CV_Error(Error::StsError,
+                "Output subscript has letters that were not encountered in the inputs");
+            }
+
+            // Push output dimention
+            // Einsum layer only has one output vector
+            dims.push_back(subscriptIndicesToDimValue[mappedIndex]);
+
+            // Reset the last input index for this subscript label
+            // given that it is seen in the output and hence can't be reduced
+            subscriptIndicesToLastInput[mappedIndex] = -1;
+            subscriptIndicesToOutputIndices[mappedIndex] = outputDimCounter++;
+        }
+    }
+    outputDims.clear();
+    outputDims.push_back(dims);
+    return result;
+}
+
+bool LayerEinsumImpl::createOutputSubsctipt() const
+{
+    // The explicit form requires no operation, as the output
+    // would have already been parsed during the input parsing process.
+    bool result = true;
+    if(explicitEquation)
+    {
+        // Ensure that the provided explicit equation includes an ellipsis if the input contains ellipses.
+        if(numOfEllipsisDims > 0)
+        {
+            if(rhs_eq.find("...") == std::string::npos)
+            {
+                CV_Error(Error::StsError,
+                "Provided output subscript does not include ellipsis while Inputs subscrits constain ellipsis");
+                result = false;
+            }
+        }
+    }
+    return result;
+}
+
+bool LayerEinsumImpl::processBroadcastedDims() const
+{
+    bool result = true;
+    // Only compute this function if ellipsis "..." was found in the equation
+    if (numOfEllipsisDims > 0){
+        CV_Error(Error::StsError, "Ellipsis are not supperted currenly");
+        result = false;
+    }
+    return result;
+}
+
+
+
+bool LayerEinsumImpl::processEquation(const String& equation, const std::vector<MatShape>& inputs) const
+{
+    bool result = true;
```

Comment:
I cannot find any line which changes result's value. Do we need it?

Reply:
Deleted!